rest/adminapitest/admin_api_test.go

```diff
@@ -1433,6 +1433,94 @@ func TestResyncStop(t *testing.T) {
 	assert.True(t, syncFnCount < 2000, "Expected syncFnCount < 2000 but syncFnCount=%d", syncFnCount)
 }
 
+// TestCorruptDbConfigHandling:
+//   - Create persistent config rest tester
+//   - Create a db with a non-corrupt config
+//   - Grab the persisted config from the bucket for purposes of changing the bucket name on it (this simulates a
+//     dbconfig becoming corrupt)
+//   - Update the persisted config to change the bucket name to a non-existent bucket
+//   - Assert that the db context and config are removed from server context and that operations on the database GET and
+//     DELETE fail with appropriate error message for user
+//   - Test we are able to update the config to correct the corrupted db config
+//   - asser the db returns to the server context and is removed from corrupt database tracking AND that the bucket name
+//     on the config now matches the rest tester bucket name
+func TestCorruptDbConfigHandling(t *testing.T) {
+	base.LongRunningTest(t)
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("Walrus/Rosmar does not support persistent config")
+	}
+	numCollections := 1
+	base.RequireNumTestDataStores(t, numCollections)
+
+	rt := rest.NewRestTester(t, &rest.RestTesterConfig{
+		CustomTestBucket: base.GetPersistentTestBucket(t),
+		PersistentConfig: true,
+		MutateStartupConfig: func(config *rest.StartupConfig) {
+			// configure the interval time to pick up new configs from the bucket to every 2 seconds
+			config.Bootstrap.ConfigUpdateFrequency = base.NewConfigDuration(2)
+		},
+	})
+	defer rt.Close()
+
+	ds := rt.CustomTestBucket.GetSingleDataStore()
+	name, ok := base.AsDataStoreName(ds)
+	require.True(t, ok)
+
+	// create db with correct config
+	resp := rt.SendAdminRequest(http.MethodPut, "/db1/", fmt.Sprintf(
+		`{"bucket": "%s", "num_index_replicas": 0, "enable_shared_bucket_access": %t, "scopes": {"%s": {"collections": {"%s":{}}}}}`,
+		rt.CustomTestBucket.GetName(), base.TestUseXattrs(), name.ScopeName(), name.CollectionName()))
+	rest.RequireStatus(t, resp, http.StatusCreated)
```

Comment:
Can you use https://github.com/couchbase/sync_gateway/blob/master/rest/utilities_testing.go#L441